File: examples/tabs/tabs.cpp
// SPDX-License-Identifier: BSD-3-Clause
// Copyright (c) 2024-2025 Darby Johnston
// All rights reserved.

#include <feather-tk/ui/App.h>
#include <feather-tk/ui/Label.h>
#include <feather-tk/ui/MainWindow.h>
#include <feather-tk/ui/ScrollWidget.h>
#include <feather-tk/ui/TabWidget.h>

#include <feather-tk/core/Format.h>
#include <feather-tk/core/String.h>

using namespace feather_tk;

FEATHER_TK_MAIN()
{
    try
    {
        // Create the context and application.
        auto context = Context::create();
        auto app = App::create(context, argc, argv, "tabs", "Tabs example");
        if (app->getExit() != 0)
            return app->getExit();

        // Create a window.
        auto window = MainWindow::create(
            context,
            app,
            "tabs",
            Size2I(1280, 960));

        // Create a tab widget.
        auto tabWidget = TabWidget::create(context);
        //tabWidget->setTabsClosable(true);
        tabWidget->setVStretch(Stretch::Expanding);
        window->setWidget(tabWidget);

        // Add tabs.
        for (size_t i = 0; i < 10; ++i)
        {
            auto label = Label::create(context, getLoremIpsum(100));
            label->setMarginRole(SizeRole::MarginSmall);
            auto scrollWidget = ScrollWidget::create(context);
            scrollWidget->setBorder(false);
            scrollWidget->setWidget(label);
            tabWidget->addTab(Format("Tab {0}").arg(i), scrollWidget);
        }

        // Show the window and run the application.
        window->show();
        app->run();
    }
    catch (const std::exception& e)
    {
        std::cout << "ERROR: " << e.what() << std::endl;
        return 1;
    }
    return 0;
}

File: python/examples/tabs.py
# SPDX-License-Identifier: BSD-3-Clause
# Copyright (c) 2024 Darby Johnston
# All rights reserved.

import feather_tk as ftk

import sys

# Create the context and application.
context = ftk.Context()
app = ftk.App(context, sys.argv, "tabs", "Tabs example")
if app.getExit() != 0:
    sys.exit(1)

# Create a window.
window = ftk.MainWindow(context, app, "tabs", ftk.Size2I(1280, 960))

# Create a tab widget.
tabWidget = ftk.TabWidget(context)
tabWidget.vStretch = ftk.Stretch.Expanding
window.setWidget(tabWidget)

# Add tabs.
for i in range(0, 10):
    label = ftk.Label(context, ftk.getLoremIpsum(100))
    label.marginRole = ftk.SizeRole.MarginSmall
    scrollWidget = ftk.ScrollWidget(context)
    scrollWidget.border = False
    scrollWidget.widget = label
    tabWidget.addTab("Tab {}".format(i), scrollWidget)

# Show the window and run the application.
window.show()
app.run()

# \bug Need to manually reset the window.
window = None
